### ISSUE TITLE
Wazuh EventLog fails if EventLog service is restarted

### ISSUE THREAD

**Issue by migruiz4** [wazuh/wazuh#3792]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.9.3-1 | Logcollector | Windows Agent | Package | Windows Server 2012 |

Hi team,

I found an issue with EventLog not being able to read any events after Eventlog is forced to stop.

When Eventlog stops, Windows automatically restarts its service, but Wazuh Eventlog enters an infinite loop with the following debug message:

```
DEBUG: Error reading event log: 6
DEBUG: Error reading event log: 6
DEBUG: Error reading event log: 6
DEBUG: Error reading event log: 6
```

I tracked this message to the code, and found the following line:

https://github.com/wazuh/wazuh/blob/68aaf6d31557106bc8b3cb58b81388644cb27ff5/src/logcollector/read_win_el.c#L487-L514

Seems like, when event log stops, it will only try to re-open it if the cause was that EventLog was cleared, but not if it was forced to stop, so will never reopen.

This bug could potentially disable Wazuh  Eventlog if something causes Eventlog to be temporarily closed.

In order to reproduce this issue, enable internal option `windows.debug=1` and execute the following command:

![imagen](https://user-images.githubusercontent.com/37381070/62319879-38f6aa00-b49f-11e9-9285-e8f72a2f03c2.png)

![imagen](https://user-images.githubusercontent.com/37381070/62319968-680d1b80-b49f-11e9-9fd9-252dae6ca6af.png)

Best regards,
Miguel

**Comment by migruiz4** [comment]
Update: This same issue happens to Eventchannel but silently.

If EventLog service is restarted, the agent will stop sending events to the manager from both Eventlog and Eventchannel.

**Comment by DaveVG1** [comment]
| Working branch |
| --- |
| [fix-eventlog-restart-3792](https://github.com/wazuh/wazuh/tree/fix-eventlog-restart-3792) |

I have fixed this issue the following way:
- For `EventLog`, I've handled the error `ERROR_INVALID_HANDLE` (that was appearing when restarting the `EventLog` service) trying to re-connect to the service. The message `DEBUG: Error reading event log: 6` still appears (about 3 times) and then it re-connects successfully.
- For `EventChannel` it was a bit trickier, as it works with a callback function when an event of this kind is triggered. I realized that when `EventLog` service was being restarted we were receiving an event with its action being `EvtSubscribeActionError`. This way, I handled this by trying to reconnect when this event was being received (waiting 5 seconds per try for a total of 3 tries as `EventLog` take some time to restart).

**Comment by chemamartinez** [comment]
Merged at #3836
